%% BioMed_Central_Tex_Template_v1.05
%%                                      %
%  bmc_article.tex            ver: 1.05 %
%                                       %


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%         <27 January 2006>           %%
%%                                     %%
%%                                     %%
%% Uses:                               %%
%% cite.sty, url.sty, bmc_article.cls  %%
%% ifthen.sty. multicol.sty		      %%
%%									   %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.pdf and the instructions for    %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\NeedsTeXFormat{LaTeX2e}[1995/12/01]
\documentclass[10pt]{bmc_article}



% Load packages
\usepackage{cite} % Make references as [1-4], not [1,2,3,4]
\usepackage{url}  % Formatting web addresses
\usepackage{ifthen}  % Conditional
\usepackage{multicol}   %Columns
\usepackage[utf8]{inputenc} %unicode support
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails
\urlstyle{rm}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %%
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %%
%%  submitted article.                         %%
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\def\includegraphic{}
\def\includegraphics{}



\setlength{\topmargin}{0.0cm}
\setlength{\textheight}{21.5cm}
\setlength{\oddsidemargin}{0cm}
\setlength{\textwidth}{16.5cm}
\setlength{\columnsep}{0.6cm}

\newboolean{publ}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                              %%
%% You may change the following style settings  %%
%% Should you wish to format your article       %%
%% in a publication style for printing out and  %%
%% sharing with colleagues, but ensure that     %%
%% before submitting to BMC that the style is   %%
%% returned to the Review style setting.        %%
%%                                              %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%Review style settings
%\newenvironment{bmcformat}{\begin{raggedright}\baselineskip20pt\sloppy\setboolean{publ}{false}}{\end{raggedright}\baselineskip20pt\sloppy}

%Publication style settings
\newenvironment{bmcformat}{\fussy\setboolean{publ}{true}}{\fussy}



% Begin ...
\begin{document}
\begin{bmcformat}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Open Data, Open Source and Open Standards in chemistry: \\ The Blue Obelisk five years on}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Ensure \and is entered between all but   %%
%% the last two authors. This will be       %%
%% replaced by a comma in the final article %%
%%                                          %%
%% Ensure there are no trailing spaces at   %%
%% the ends of the lines                    %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\author{
  Samuel E Adams$^8$%
  \email{Samuel E Adams - sea36@cam.ac.uk}
  \and
  Jonathan Alvarsson$^1$%
  \email{Jonathan Alvarsson - jonathan.alvarsson@farmbio.uu.se}
  \and
  Igor V Filippov$^2$%
  \email{Igor V Filippov - igorf@helix.nih.gov}
  \and
  Rajarshi Guha$^2$%
  \email{Rajarshi Guha - guhar@mail.nih.gov}
  \and
  Robert Hanson$^3$%
  \email{Robert Hanson - hansonr@stolaf.edu}
  \and
  Marcus D Hanwell$^{4}$%
  \email{Marcus D Hanwell - marcus.hanwell@kitware.com}
  \and
  Geoffrey R Hutchison$^5$%
  \email{Geoffrey R Hutchison - geoffh@pitt.edu}
  \and
  Karol M Langner$^6$%
  \email{Karol M Langner - langnerkm@chem.leidenuniv.nl}
  \and
  Daniel M Lowe$^8$%
 \email{Daniel Lowe - dl387@cam.ac.uk}
  \and
  Peter Murray-Rust$^8$%
  \email{Peter Murray-Rust - pm286@cam.ac.uk}
  \and
  Noel M O'Boyle\correspondingauthor$^9$%
  \email{Noel M O'Boyle\correspondingauthor - n.oboyle@ucc.ie}
  \and
  Dmitry Pavlov$^{10}$%
  \email{Dmitry Pavlov - dpavlov@ggasoftware.com}
  \and
  Ola Spjuth$^1$%
  \email{ola.spjuth@farmbio.uu.se}
  \and
  Christoph Steinbeck$^{11}$%
  \email{Christoph Steinbeck - steinbeck@ebi.ac.uk}
  \and
  Adam L Tenderholt$^{12}$%
  \email{Adam L Tenderholt - adamlt82@u.washington.edu}
  \and
  Kevin J Theisen$^{13}$%
  \email{Kevin J Theisen - kevin@ichemlabs.com}
  \and
  Egon L Willighagen$^{14}$%
  \email{Egon L Willighagen - egon.willighagen@ki.se}
}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\address{%
    \iid(1)Department of Pharmaceutical Biosciences, Uppsala University, Box 591, 751 24 Uppsala, Sweden \\
    \iid(2)NIH Center for Translational Therapeutics, 9800 Medical  Center Drive, Rockville, MD 20878 \\
    \iid(3)... \\
    \iid(4)Kitware, Inc., 28 Corporate Drive, Clifton Park, NY 12065, USA \\
    \iid(5)Department of Chemistry, University of Pittsburgh, 219 Parkman Avenue, Pittsburgh, PA 15260, USA \\
    \iid(6)Gorlaeus Laboratories, Leiden University, 2333 CC Leiden, The Netherlands \\
    \iid(8)Unilever Centre for Molecular Sciences Informatics, Department of Chemistry, University of Cambridge, Lensfield Road, CB2 1EW, UK \\
    \iid(9)Analytical and Biological Chemistry Research Facility, Cavanagh Pharmacy Building, University College Cork, College Road, Cork, Co. Cork, Ireland \\
    \iid(10)GGA Software Services LLC, 41 Nab. Chernoi rechki 194342, Saint Petersburg, Russia \\
    \iid(11)Cheminformatics and Metabolism Team, European Bioinformatics Institute (EBI), Wellcome Trust Genome Campus, Hinxton, Cambridge, UK \\
    \iid(12)Department of Chemistry, University of Washington, Seattle, WA 98195 \\
    \iid(13)iChemLabs, 200 Centennial Ave., Suite 200, Piscataway, NJ 08854, USA \\
    \iid(14)Division of Molecular Toxicology, Institute of Environmental Medicine, %
        Nobels vaeg 13, Karolinska Institutet, 171 77 Stockholm, Sweden
}%

\maketitle

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the Instructions for       %%
%% authors on http://www.biomedcentral.com  %%
%% and change the section headings          %%
%% accordingly.                             %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\begin{abstract}
        % Do not use inserted blank lines (ie \\) until main body of text.
        \paragraph*{Background:} The Blue Obelisk movement was established in 2005 as a
		response to the lack of Open Data, Open Standards and
		Open Source (ODOSOS) in chemistry. It aims to
make it easier to carry out chemistry research by
promoting interoperability between chemistry software, encouraging cooperation
between Open Source developers, and developing community resources and
Open Standards.

        \paragraph*{Results:} This
		contribution looks back on the past 5 years and surveys progress and
		remaining challenges in the areas of Open Data, Open Standards, and
    Open Source in chemistry.

        \paragraph*{Conclusions:} We show that the Blue Obelisk has been very successful
		in bringing together researchers and developers with common interests
		in ODOSOS, leading to development of many useful resources freely
		available to the chemistry community. But how best to engage with the
		wider chemistry community outside of the Blue Obelisk remains an open
		question.
\end{abstract}



\ifthenelse{\boolean{publ}}{\begin{multicols}{2}}{}




%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the instructions for       %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and change the section headings          %%
%% accordingly.                             %%
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%




%%%%%%%%%%%%%%%%
%% Background %%
%%
\section*{Background}
The Blue Obelisk movement was established in 2005 at the
229\textsuperscript{th} National Meeting of the American Chemistry
Society as a response to the lack of Open Data, Open Standards and
Open Source (ODOSOS) in chemistry. While other scientific disciplines
such as physics, biology and astronomy (to name a few) were embracing
new ways of doing science and reaping the benefits of community
efforts, there was little if any innovation in the field of chemistry
and scientific progress was actively hampered by the lack of access to
data and tools.
Since 2005 it has become evident that a good amount of development in open
chemical information is driven by the demands of neighbouring
scientific fields. In many areas in biology, for example, the importance of
small molecules and their interactions and reactions in biological systems
has been realised. In fact, one of the first free and open databases and ontologies
of small molecules was created as a resource about chemical structure and nomenclature
by biologists.\cite{DeMatos:2009p3839}

The formation of the Blue Obelisk group is somewhat unusual in that it
is not a funded network, nor does it follow the industry consortium
model. Rather it is a grassroots organisation, catalysed by an initial
core of interested scientists, but with membership open to all who
share one or more of the goals of the group:
\begin{itemize}
\item {\bf Open Data in Chemistry.} One can obtain all scientific data in the public domain when
wanted and reuse it for whatever purpose.
\item {\bf Open Standards in Chemistry.} One can find visible community mechanisms for
protocols and communicating information. The mechanisms for creating
and maintaining these standards cover a wide spectrum of human
organisations, including various degrees of consent.
\item {\bf Open Source in Chemistry.} One can use other people's code without further
permission, including changing it for one's own use and distributing
it again.
\end{itemize}

Note that while some may advocate also for Open Access to
publications, the Blue Obelisk goals (ODOSOS) focus more on the
availability of the underlying scientific data, standards (to exchange
data), and code (to reproduce results). All three of these goals stem
from the fundamental tenants of the scientific method for data sharing
and reproducibility.

The Blue Obelisk was first described in the CDK News \cite{CDKNewsBO} and
later as a formal paper by Guha et al.\cite{Guha2006} in
2006. This contribution looks back on the past 5 years and surveys
progress and remaining challenges in the areas of Open Data, Open
Source, and Open Standards in chemistry.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Main section %%
%%

\section*{Scope}
The Blue Obelisk covers many areas of chemistry and chemical resources
used by neighbouring disciplines ({\it e.g.} biochemistry, materials
science). Many of the efforts relate to cheminformatics (the scope of
this journal) and we believe that many of the publications in {\it J.
Cheminf.} could be completely carried out using Blue Obelisk resources
and other Open Source chemical tools. The importance of this is that for the
first time it would allow reviewers, editors and readers to validate
assertions in the journal and also to re-run and re-analyse parts of
the calculation.

However, Blue Obelisk software and data is used outside
cheminformatics and certainly in the five main areas that, for
example, Chemical Markup Language (CML) \cite{murray-rust_chemical_1999} supports:

\begin{itemize}
\item{1.}{\bf Molecules:} This is probably the largest area for Blue
Obelisk software and data, and is reflected by many programs that
visualise, transform, convert formats and calculate properties. It is
almost certain that any file format currently in use can be processed by Blue
Obelisk software and that properties can be calculated for most (organic
compounds).
\item{2.}{\bf Reactions:} Blue Obelisk software can describe the
semantics of reactions and provide atom-atom matching and analyse
stoichiometric balance in reactions.
\item{3.}{\bf Computational chemistry:} Blue Obelisk software can
interpret many of the current output files from calculations and
create input for jobs. The Quixote project (see below and elsewhere in
this issue) shows that Open Source approaches based on Blue Obelisk
resources and principles are increasing the availability and
re-usability of computational chemistry.
\item{4.}{\bf Spectra:} 1-D spectra (NMR, IR, UV etc.) are fully
supported in Blue Obelisk offerings for conversion and display. There
is a limited amount of spectral analysis but the software gives a
platform on which it should be straightforward to develop spectral
annotation and manipulation. However, currently the Blue Obelisk lacks
support for multi-dimensional NMR and multi-equipment spectra
({\it e.g.} GC-MS).
\item{5.}{\bf Crystallography:} The Blue Obelisk software supports the
bi-directional processing of crystal structure files (CIF) and also
solid-state calculations such as plane-waves with periodic boundary
conditions. There is considerable support for the visualisation of
both periodic and aperiodic condensed objects.
\end{itemize}

Many of the current operations in installing and running chemical
computations and using the data are integration and customisation
rather than fundamental algorithms. It is very difficult to create
universal platforms that can be distributed and run by a wide range of
different users, and in general, the Blue Obelisk deliberately does
not address these. Our approach is to produce components that can be
embedded in many environments, from stand-alone applications to web
applications, databases and workflows.
We believe that a chemical laboratory with reasonable access to common
software engineering techniques should be able to build customised applications
using Blue Obelisk components and standard infrastructure such as
workflows (KNIME or Taverna) and databases (MySQL, CouchDB etc.).
Where the Blue Obelisk itself produces data resources (NMRShiftDB,
CrystalEye, Quixote etc.) they are normally done with Open components
so that the community can, if necessary, replicate them.

Much of the impetus behind Blue Obelisk software is to create an
environment for chemical computation (including cheminformatics) where
all of the components, data, specifications, semantics, ontology and
software are Openly visible and discussable. The largest current uses
by the general chemical community are in authoring, visualisation and
cheminformatics calculations but we anticipate that this will shortly
extend into mainstream computational chemistry and solid-state.
Although many of the authors are employed as research scientists,
there are also several people who contribute in their spare time and
we anticipate an increasing value and use of the Blue Obelisk in
education at all levels.


\section*{Open Source}

\subsection*{Cheminformatics toolkits}

Open Source toolkits for cheminformatics have now existed for nearly
ten years. During this period, some toolkits were developed from
scratch in academia, whereas others were made Open Source by releasing in-house
codebases under liberal licenses. When the Blue Obelisk was
established five years ago, the primary toolkits under active development
were the Chemistry Development Kit (CDK)
\cite{Steinbeck2003, Steinbeck2006}, Open Babel \cite{WebOpen Babel},
and JOELib \cite{WebJOELib}. Of these, both the CDK and Open Babel
continue to be actively developed.

The CDK project has been under regular development over the last five
years. Several feature have been implemented ranging from core
components such as an extensible SMARTS matching system and a new graph
(and
subgraph) isomorphism method \cite{smsd}, to more application oriented
components such as 3D pharmacophore searching and matching, and a variety
of structural-key and hashed fingerprints. In addition, there have
been a number of second generation tools developed on top of the CDK
(see below). As well as the use of the CDK in various tools, it has been deployed in
the form of web services \cite{Dong:2007aa} and has formed the basis
of a variety of web applications.

Since 2006, major new features of Open Babel include 3D structure
generation and 2D structure-diagram generation, UFF and MMFF94
forcefields, and significantly expanded support for computational
chemistry calculations. In addition, a major focus of Open Babel development
has been to provide for accurate conversion and representation in
areas of stereochemistry, kekulisation, and canonicalisation. The
project has also grown, in terms of new contributors, new support from
commercial companies, and second-generation tools applying Open Babel
to a variety of end-user applications, from molecular editors to
chemical database systems.

Two new Open Source cheminformatics toolkits have appeared since the
original paper. In 2006 Rational Discovery, a cheminformatics service
company (since closed down), released RDKit \cite{WebRDKit} under the
BSD License. This is a C++ library with Python and (more recently)
Java bindings. RDKit is actively developed and includes
code donated by Novartis. Recent developments include the Java
bindings, as well as performance improvements for its database
cartridge.

More recently, GGA Software Services
(a contract programming company)
released the Indigo toolkit \cite{WebIndigo} and associated software
in 2009 under the GPL. Indigo is a C++ library with
high-level wrappers in C, Java, Python, and the .NET
environment. Like RDKit and other toolkits, Indigo provides support for
tetrahedral and cis-trans stereochemistry, 2D coordinate generation,
exact/substructure/SMARTS matching, fingerprint generation, and
canonical SMILES computation.
It also provides some less common functionality, like matching
tautomers and resonance substructures, enumeration of subgraphs,
finding maximum common substructure of $N$ input structures, and
enumerating reaction products.

\subsection*{Second-generation tools}

Although feature-rich and robust cheminformatics toolkits are useful
in and of themselves, they can also be seen as providing a base layer
on which additional tools and applications can be built. This is one
of the reasons that cheminformatics toolkits are so important to the
open source `ecosystem'; their availability lowers the barrier for the
development of a `second generation' of chemistry software that no
longer needs to concern itself with the low-level details of
manipulating chemical structures, and can focus on providing
additional functionality and ease-of-use. Although a wide range of
chemistry software has been built using Blue Obelisk
components (see for example, the ``Related Software'' link on the Open
Babel website,\cite{WebOBRelated} listing over 40 projects as of this writing, 
or ``Software using CDK'' at the CDK website), in this
section we focus on second-generation tools which themselves have been
developed by members of the Blue Obelisk. 

Bioclipse~\cite{Spjuth:2007fk} (v2.4 released in Aug 2010) and Avogadro
\cite{WebAvogadro} (v1.0 in Oct 2009) are two examples of such software, based
on the CDK and Open Babel, respectively. Bioclipse (Figure~1) is an award-winning
molecular workbench for life sciences
that wraps cheminformatics functionality behind user-friendly interfaces and
graphical editors while Avogadro (Figure~2) is a 3D molecular editor and viewer aimed at
preparing and analysing computational chemistry calculations. Both
projects are designed to be extended or scripted by users through
the provision of a plugin architecture and scripting support (using
Bioclipse Scripting Language~\cite{Bioclipse2}, or Python in the case
of Avogadro). An interesting aspect of both Avogadro and Bioclipse is
that they share some developers with the underlying toolkits and this
has driven the development of new features in the CDK and Open Babel.

Both products in turn act as extensible platforms for other
software. Bioclipse, for example is used by software
such as Brunn~\cite{Alvarsson:2011fk}, a laboratory information system for
microplate based high-throughput screening. Brunn provides a graphical interface
for handling different plate layouts and dilution series and can automatically
generate dose response curves and calculate IC$_{50}$-values. Avogadro
is used by Kalzium~\cite{WebKalzium}, a periodic table and chemical editor in KDE,
and XtalOpt~\cite{WebXtalOpt}\cite{Lonie2011}, an evolutionary
algorithm for crystal structure prediction. XtalOpt provides a
graphical interface using Avogadro and submits calculations using a
range of solid-state simulation software to predict stable polymorphs.

A final example of second-generation Blue Obelisk software is the 
AMBIT2~\cite{WebAMBIT} software, a GUI that facilitates registration
of chemicals for the REACH EU directive on toxicity, and which is
based on the CDK.

\subsection*{Computational chemistry analysis}

Another area where the Blue Obelisk has had a significant impact in the
past five years is in supporting
quantum chemistry calculations and in interpreting their results.
Electronic structure calculations have a long tradition in the
chemistry community and a variety of programs exist, mostly
proprietary software with the notable exception of
NWChem~\cite{NWChem}.
However, since each program uses different input formats, and the
the output formats vary widely (sometimes even varying between different versions
of the same software), preparing calculations and automatically extracting
the results is problematic. 

Avogadro has already been mentioned as a GUI for preparing calculations.
It uses Open Babel to read the output of several electronic structure
packages. Avogadro generates input files on the fly in response to user
input on forms, as well as allowing inline editing of the files before
they are saved to disk. It also features intuitive syntax highlighting
for GAMESS input files, allowing expert users to easily spot mistakes
before saving an input file to disk.

In addition to this, significant development of new parsing routines took
place in an Avogadro plugin to read in basis sets and electronic structure
output in order to calculate molecular orbital and electron density grids.
This code was written to be parallel, using desktop shared memory parallelism
and high level APIs in order to significantly speed up analysis. Most of this
code was recently separated from the plugin, and released as a BSD licensed
library, OpenQube, which is now used by the latest version of
Avogadro.
Jmol (see below)
can also depict computational chemistry results including molecular orbitals.

In 2006, the Blue Obelisk project cclib~\cite{cclib} was established
with the goal of parsing the output from computational chemistry
programs and presenting it in a standard way so that further analyses
could be carried out independently of the quantum package used.
cclib is a Python library, and the current version (version 1.0.1)
supports 8 different computational chemistry codes and extracts over
30 different calculated attributes. Two related BO projects build upon 
cclib. GaussSum~\cite{WebGaussSum},
is a GUI that can monitors the progress of SCF and geometry convergences, 
and can plot predicted UV/Vis absorption and infrared spectra from 
appropriate logfiles containing energies and oscillator strengths for easy 
comparison to experimental data. QMForge~\cite{WebQMForge} provides 
a GUI for various electronic structure analyses such as Frenking's charge 
decomposition analysis~\cite{Frenking} and Mulliken or C-squared analyses
on user-defined molecular fragments. QMForge also provides a rudimentary
Cartesian coordinate editor allowing molecular structures to be saved via Open Babel.

The Quixote project epitomises the full use of the Blue Obelisk
software and is described in detail in a sibling
article~\cite{QuixoteSameIssue}. Here we observe that it is possible
to convert legacy chemistry file formats
of all sorts into semantic chemistry and extract
those parts which are suitable for input to computational chemistry
programs. This chemistry is then combined with
generic concepts of computational chemistry ({\it e.g.} strategy,
machine resources, timing, accuracy etc.) into the
legacy inputs for a wide range of programs. Quixote itself follows
Blue Obelisk principles in that it does not manage
the submission and monitoring of jobs but resumes action when the jobs
have been completed, and then applies a range
of parsing and transformation tools to create standardised semantic
chemical content. A major feature of Quixote is
that it requires all concepts to validate against dictionaries and the
process of parsing files necessarily generates
communally-agreed dictionaries, which represent an important step
forward in the Open specifications for Blue Obelisk.
When widely-deployed, Quixote will advertise the value of Open
community standards for semantics to the world.

The Quixote project is not dependent on any particular technology,
other than the representation of computational
chemistry in CML and the management of semantics through CML
dictionaries. At present, we use JUMBO-Converters~\cite{JUMBO-Convertors} for most
 of the semantic conversion, Lensfield2~\cite{Lensfield2} for the workflow and Chempound
(chem\#) \cite{Chempound} to store and disseminate the results.

\subsection*{Web applications}

While desktop software has composed the majority of scientific tools
since the computer was introduced, the internet continues to change
how applications and content are distributed and presented. The web
presents new opportunities for scientists as it
an open and free medium to distribute scientific knowledge, ideas and
education. Web applications are software that runs within the browser,
typically implemented in Java or JavaScript.
Recently, a new version of the HTML
specification, HTML5, defines a well-developed framework
for creating native web applications in JavaScript and opens up
new possibilities for visualising chemical data.

Jmol, the interactive 3D molecular viewer, is one of the most widely used
chemistry applets, and indeed has
seen widespread use in other fields such as biology and 
even mathematics (it is used for 3D depiction of mathematical
functions in the Sage Mathematics Projects~\cite{WebSage}). It is implemented
in Java, and has gone from being a ``Rasmol/Chime'' replacement to a fully fledged molecular
visualisation package, including full support for crystallography~\cite{Hanson2010},
display of molecular orbitals from standard basis set/coefficient data,
the inclusion of dynamic minimisation using the UFF force field, and
a full implementation of Daylight SMILES and SMARTS, with extensions to
conformational and biomolecular substructure searching (Jmol
BioSMARTS).

In 2009, iChemLabs released the ChemDoodle Web Components
library~\cite{ChemDoodleWeb} under the GPL v3 license (with a
liberal HTML exception). This library is completely implemented in JavaScript
and uses HTML5 to allow the scientist
to present publication quality 2D and 3D graphics (see Figure~3) and animations for
chemical structures, reactions and spectra. Beyond graphics, this tool
provides a framework for user interaction to create dynamic
applications through web browsers, desktop platforms and mobile
devices such as the iPhone, iPad and Android devices.

\subsection*{The business end}

% We should cite Craig James
% Care and Feeding of FIOSS: http://www.moonviewscientific.com/essays/software_lifecycle.htm

Open Source provides a unique opportunity for commercial organisations to work with the
cheminformatics community. Traditional business models rely on monetisation of source
code, causing companies to repeat work done by other companies. This
model is sometimes combined
with a free (gratis) model for people working at academic institutes, to increase adoption
and encourage contributions from academics. This solution defines the return on investment as the IP on the software,
but has the downside of investment losses due to duplication of
software and method development, which become visible when proprietary companies
merge. Some authors have argued that in the chemistry field
few contributors are available to volunteer time
to improve codes and IP considerations may prevent contributions from
industry~\cite{Stahl:2005fk}. If true, this would hamper
adoption of Open Source and Open Data in chemistry, and greatly slow
the growth of projects such as those in the Blue Obelisk.

The Blue Obelisk community, however, takes advantage of the fact that much of the investment needed
for development is either paid for by academic institutes and funding
schemes, or by
volunteers investing time and effort. In return, contributors get full
access to the source code, and the Open Source licensing ensures
that they will have access any time in the future. In this way, the
license functions as a social
contract between everyone to arrange an immediate return on investment. Effectively,
this approach shares the burden of the high investment in having to develop cheminformatics
software from scratch, allowing researchers and commercial partners alike to focus
on their core business, rather than the development of prerequisites.
In the case of the Blue Obelisk, the rich
collection of Open Source cheminformatics tools provided
greatly reduces investment up front for new companies in the
cheminformatics market. Such advantages have also been noted in the
drug discovery
field \cite{DeLano:2005uq}\cite{Munos:2006vn}\cite{Geldenhuys:2006kx}.

The use of Open Standards allows everyone to select those Blue Obelisk components
they find most useful, as they can easily replace one component with another providing
the same functionality, taking advantage that they use the same standards for,
for example, data exchange. This way, licensing issues are becoming a marginal
problem, allowing companies to select a license appropriate for their business
model. This too, allows a company to create a successful product with significantly
reduced cost and effort.

At the time of writing there are many commercial companies development chemistry
solutions around Open Source cheminformatics components provided by the Blue Obelisk
community. Examples of such companies include iChemLabs, IdeaConsult, Wingu, Silicos,
GenettaSoft, hBar, and Inkspot Science. Some of these merely use components, but several
actively contribute back to the Blue Obelisk project they use, or donate new
Open Source cheminformatics projects to the community.

For example, iChemLabs released the ChemDoodle Web Components library under the GPL v3
license, based on the upcoming HTML5 Open Standard. It allows making web and mobile
interfaces for chemical content. The project is already being adopted by others,
including iBabel~\cite{iBabel}, ChemSpotlight~\cite{chemspotlight} and the RSC ChemSpider~\cite{chemspider_chemdoodle}.

Silicos has release several Open Source utilities based on
Open Babel, such as Pharao, a tool for pharmacophore searching,
Sieve for filtering molecular structure by molecular property,
Stripper for removing core scaffold structures from a molecule
set, and Piramid for molecular alignment using shape determined
by the Gaussian volumes as a descriptor. Additionally,
contributions have been made to the Open Babel project itself.

Other companies use Blue Obelisk components and contribute patches,
smaller and larger. For example, IXELIS donated the isomorphism
code in the CDK, eMolecules donated canonicalisation code to
Open Babel, and AstraZeneca contributed code to the CDK for
signatures. This is just a very minor selection, and the author
is encouraged to contact the individual Blue Obelisk projects
for a detailed list.

In May 2011, a Wellcome Trust Workshop on
Molecular Informatics Open Source Software (MIOSS) explored the role of
Open Source in industrial laboratories and companies as well as
academia (several of
the presenters are among the authors of this paper).
The meeting identified that Open Source software was extremely valuable to
industry not just because it is available for free,
but because it allows the validation of source code, data and
computational procedures. Some 
of the discussion was on business models or other ways to maintain 
development of Open Source software on which a business relied.
Companies are concerned about training and support
and, in some cases, product liability.  There are difficulties
for software for which there is no formal
transaction other than downloading and agreeing to license terms.
One anecdote concerned a company
that wished to donate money to an Open Source
project but could not find a mechanism to do so.

Industry participants also pointed out that 
there is a considerable amount of
contribution-in-kind from industry, both from
enhancements to software and also the development of completely new
software and toolkits. Companies are now finding it easier to create
mechanisms for releasing Open Source software
without violating confidentiality or incurring liability.
A phrase from the meeting
summed it up: ``The ice is beginning to melt'', signifying that we can
expect a rapid increase in industry's interest
in Open Source.

\subsection*{Converting chemical names and images to structures}

The majority of chemical information is not stored in machine-readable
formats, but rather as chemical names or depictions. The OSRA and OPSIN
projects focus on extracting chemical information from these sources.
Such software plays a particularly important role for data mining the
chemical literature, including patents and theses.

Optical Structure Recognition Application (OSRA) \cite{WebOSRA} was started
in early 2007 with the goal to create the first free and open source
tool for extraction and conversion of molecular images into SMILES and
SD files. From the very beginning the underlying philosophy was to integrate
existing open source libraries and to avoid ``reinventing the wheel''
wherever possible. OSRA relies on a variety of open source components:
Open Babel for chemical format
conversion and molecular property calculations, GraphicsMagick for image
manipulation, Potrace for vectorisation, GOCR and OCRAD for optical
character recognition. The growing importance of image
recognition technology can be seen in the fact that
only a few years ago there was only one widely available software
package for chemical structure recognition -  CLiDE (commercially
developed at Keymodule, Ltd), but today there are as many as seven
available programs.

OPSIN (Open Parser for Systematic IUPAC
Nomenclature)~\cite{lowe_chemical_2011} focuses instead on interpreting chemical names.
The chemical name is the oldest form of communication used to
describe chemicals, predating
even the knowledge of the atomic structure of compounds.
Chemical names are abundant in the scientific
literature and encode valuable structural information.
Through successive books of
recommendations~\cite{iupac_nomenclature_1979, iupac_guide_1993},
IUPAC has tried to codify and to an extent standardise naming practices.
OPSIN aims to make this abundance of
chemical names machine readable by translating them to SMILES, CML or
InChI. The program is based around the use of a regular grammar to
guide tokenisation and parsing of chemical names, followed by
step-wise application of nomenclature rules. It is able to offer
fast and precise conversions for the majority of names using IUPAC
organic nomenclature, and is available as a web service, Java
library and standalone application for maximum interoperability.

\subsection*{Chemical database software}

Registration, indexing and searching of chemical structures in
relational databases is one of the core areas of cheminformatics.
A number of structure registration systems have been published in the last five years, exploiting the fact that
Open Source cheminformatics toolkits such as Open Babel and the CDK
are available.
OrChem~\cite{RijnbeekS10}, for example, is an open source extension for the Oracle 11G database that
adds registration and indexing of chemical structures to support fast
substructure and similarity searching. The cheminformatics
functionality is provided by the CDK. OrChem
provides similarity searching with response times in the order of
seconds for databases with millions of compounds, depending on a given
similarity cut-off. For substructure searching, it can make use of
multiple processor cores on today's powerful database servers to
provide fast response times in equally large data sets.

Besides the traditional and proven relational database approach with
added chemical features (`cartridges'), there is growing
interest in tools and approaches based on the web philosophy and
practice. Several groups are experimenting with 
the Resource Description Framework (RDF) language
on the assumption that generic high-performance solutions will appear.
RDF allows everything to be described by
URIs (data, molecules, dictionaries, relations). The Chempound
system~\cite{Chempound},
as deployed in
Quixote and elsewhere, is an RDF-based approach to chemical structures
and compounds and their properties. For small
to medium-sized collections (such as an individual's calculations or
literature retrieval), there are many RDF tools
(e.g. SIMILE, Apache Jena) which can operate in machine memory and provide
the flexibility that RDF offers. For larger
systems, it is unclear whether complete RDF solutions (e.g. Virtuoso)
will be satisfactory or whether a hybrid system
based on name-value pairs (e.g. CouchDB, MongoDB) will be sufficient.

  \subsection*{Collaboration and interoperability}

One of the successes of the Blue Obelisk has been to bring developers
together from different Open Source chemistry projects so that they
look for opportunities to collaborate rather than compete, and to
leverage work done by other projects to avoid duplication of effort.
As an example of this, when in March 2008 the Jmol development team
were looking to add support for energy minimisation, rather than
implement a forcefield from scratch they ported the UFF forcefield~\cite{Rappe:1992um}
implementation from Open Babel to Jmol. This code enables Jmol to
support 2D to 3D conversion of structures (through energy
minimisation). In a similar manner, efficient Jmol code for atom-atom rebonding
has been ported to the CDK. Figure~4 shows the collaborative nature of
software developed in the Blue Obelisk, as one project builds on
functionality provided by another project.

Another collaborative initiative between Blue Obelisk projects was the establishment in May 2008 of
the ChemiSQL project. This brought together the developers of several
open source chemistry database cartridges (PgChem~\cite{WebPgChem},
MyChem~\cite{WebMyChem}, OrChem~\cite{RijnbeekS10} and
more recently Bingo~\cite{WebBingo}) with a view to making their database APIs more
similar and collaborating on benchmark datasets for assessing
performance. For two of these projects, PgChem and MyChem, which are both based on
Open Babel, there is the additional possibility of working together on a shared
codebase.

In the area of cheminformatics toolkits, two of the existing toolkits
Open Babel and RDKit are planning to work together on a common
underlying framework called MolCore~\cite{WebMolCore}. This project is still in the
planning stage, but if it is a success it will mean that the the two
libraries will be interoperable (while retaining their existing focus)
but also that the cost of maintaining the code will be shared among
more developers, freeing time for the development of new features.

One of the goals of the Blue Obelisk is to promote interoperability in chemical
informatics. When barriers exist to moving chemical data between
different software, the community becomes fragmented and there is
the danger of vendor lock-in (where users are constrained to using
a particular software, a situation which puts them at a
disadvantage). This applies as much to Open Source software as to
proprietary software. Cinfony is a project (first release in May 2008)
whose goal is to tackle this problem in the area of cheminformatics
toolkits \cite{OBoyleCinfony2008}.
It is a Python library that enables Open Babel, the CDK, and RDKit
(and shortly, Indigo and OPSIN) to
be used using the same API; this makes it easy, for example, to read a
molecule using Open Babel, calculate descriptors using the CDK and
create a depiction using RDKit.

Another way through which interoperability of Blue Obelisk projects
has been promoted and developed is through integration into
workflow software such as Taverna~\cite{Hull:2006p60} and
KNIME~\cite{WebKNIME} (both open source).
Such software makes it easy to automate recurring
tasks, and to combine analyses or data from a variety of different software
and web services.
A combination of the Chemistry Development Kit and Taverna, for instance, was
reported in 2010~\cite{Kuhn:2010p4001}. 
In the case of KNIME, it comes with built-in basic collection of CDK-based and
Open Babel-based nodes, while other nodes for the RDKit and Indigo are
available from KNIME's ``Community Updates'' site.


\section*{Open Standards}

\subsection*{Chemical Markup Language, CML}

Chemical Markup Language (CML) is discussed in several articles in this
issue, and a brief summary here re-iterates that it is designed
primarily to create a validatable semantic representation for chemical
objects. The five main areas (molecules, reactions, computational
chemistry, spectra and solid-state (see above)) have now all been
extensively deployed and tested. CML can therefore be used as a
reference for input and output for Blue Obelisk software and a means
of representing data in Blue Obelisk resources.

CML, being an XML application, can inter-operate with other markup
languages and in particular XHTML, SVG, MathML, docx and more
specialised applications such as UnitsML and GML (geosciences). We
believe that it would be possible using these languages to encode
large parts of, say, first year chemistry text books in XML.
Similarly, it is possible to create compound documents with word
processing or spreadsheet software that have inter-operating text,
graphics and chemistry (as in Chem4Word). Being a markup language, CML
is designed for re-purposing, including styling, and therefore a
mixture of these languages can be used for chemical catalogues,
general publications, logbooks and many other types of document in the
scientific process.

CML describes much of its semantics through conventions and
dictionaries, and the emerging ecosystem (especially in computational
chemistry) is available as a semantic resource for many of the
applications and specifications in this article.


  \subsection*{InChI}

The IUPAC InChI identifier is a non-proprietary and unique identifier
for chemical substances designed to enable linking of diverse data
compilations. Prior to the development of the InChI identifier chemical
information systems and databases used a wide variety of (generally
proprietary) identifiers, greatly limiting their interoperability.
Although its development predates the Blue Obelisk, software such as Open
Babel has included InChI support since 2005, and support for InChI in
Indigo is due in 2011.

Since the official InChI implementation is in C, it is difficult to
access from the other widely used language for cheminformatics
toolkits, Java. Early attempts to generate InChI identifiers from
within Java involved programatically launching the InChI executable
and capturing the output, an approach that was found to be fairly
unreliable and broke the 'write once, run anywhere' philosophy
of Java.  The Blue Obelisk project JNI-InChI~\cite{WebJNIInChI}
was established in 2006 to solve this problem by using the Java Native
Interface framework to provide transparent access to the InChI
library from within Java and other Java Virtual Machine (JVM) based
languages, supporting the wider adoption of
this standard identifier by the chemistry community.

The Java Native Interface framework provides a mechanism for code
running inside the JVM, to place calls to libraries written in languages
such as C, C++ and Fortran, and compiled into native, machine specific,
code. JNI-InChI provides a thin C wrapper, with corresponding Java code,
around the IUPAC InChI library, exposing the InChI library's functionality
to the JVM.  To overcome the need to have the correct InChI library pre-installed
on a system, JNI-InChI comes with a variety of precompiled native binaries and
automatically extracts and deploys the correct one for the detected operating
system and architecture. The JNI-InChI library comes with native binaries
supporting a range of operating systems and architectures; the current version
has binaries for 32- and 64-bit Windows, Linux and Solaris, 64-bit FreeBSD
and 64-bit Intel-based Mac OS X - a number of which are not supported by the
original IUPAC distribution of InChI.

The JNI-InChI project has matured to support the full range of
functionality of the InChI C library: structure-to-InChI, InChI-to-structure,
AuxInfo-to-structure, InChIKey generation, and InChI and InChIKey validation.
JNI-InChI provides the InChI functionality for a number of Open Source projects,
including the Chemistry Development Kit, Bioclipse and CMLXOM/JUMBO, and is
also used by commercial applications and internally in a number of companies.
Through its widespread use and Open Source development model, a number of
issues in earlier versions of the software have been identified and resolved,
and JNI-InChI now offers a robust tool for working with
InChIs in the JVM.


    \subsection*{OpenSMILES}

One of the most widely used ways to store chemical structures is the
SMILES format (or SMILES string). This is a linear notation developed
by Daylight Information Systems that describes the connection table
of a molecule and may optionally encode chirality. Its popularity
stems from the fact that it is a compact representation of the
chemical structure that is human readable and writable, and is
convenient to manipulate (e.g. to include in spreadsheets, or copy
from a web page).

Despite its widespread use, a formal
definition of the language did not exist beyond Daylight's SMILES
Theory Manual and tutorials. This caused some confusion in the
implementation and interpretation of corner cases, for example the
handling of cis/trans bond symbols at ring closures. In 2007, Craig
James (eMolecules) initiated work on the OpenSMILES specification, a
complete specification of the SMILES language as an Open Standard
developed through a community process. The specification is largely
complete and contains guidelines on reading SMILES, a formal
grammar, recommendations on standard forms when writing SMILES, as
well as proposed extensions.

\subsection*{QSAR-ML}
The field of QSAR has long been hampered by the lack of open
standards, which makes it difficult to share and reproduce descriptor
calculations and analyses. QSAR-ML was recently proposed as an open
standard for exchanging QSAR datasets~\cite{Spjuth:2010uq}. A dataset
in QSAR-ML includes the chemical structures (preferably described in
CML) with InChI to protect integrity, chemical descriptors linked to
the Blue Obelisk Descriptor Ontology~\cite{bodo}, response values,
units, and versioned descriptor implementations to allow for
integrating several descriptor software in the same calculation.
Hence, a dataset described in QSAR-ML is completely reproducible. To
allow for easy setup of QSAR-ML compliant datasets, a plugin for
Bioclipse was created with a graphical interface for
setting up QSAR datasets and performing calculations. Descriptor
implementations are available from the CDK and JOELib, as well
as via remote web services such as XMPP~\cite{Wagener:2009uq}.

\subsection*{Remaining challenges}

A core requirement for chemical structure databases and chemical
registration systems in general is the notion of structure
standardisation.  That is,  for a given input structure, multiple
representations should be converted to one canonical form.
Structure canonicalisation routines partially address this aspect,
converting multiple alternative topologies to a single canonical
form. However, the problem of standardisation is broader than just
topological canonicalisation. Features that must be considered include
\begin{itemize}
\item topological canonicalisation
\item handling of charges
\item tautomer enumeration and canonicalisation
\item normalisation of functional groups
\end{itemize}
Currently, most of the individual components of a `standardisation
pipeline' can be implemented using BO tools. The larger problem is
that there is no agreed upon list of steps for a standardisation
process. While some specifications have been published (e.g., PubChem)
and some standardisation services and tools are available (for
example, PubChem
provides an online service to standardise
molecules~\cite{WebPubChemStandardizer})
each group has their own set of rules. A
common reference specification for standardisation would be of immense
value in interoperability between structure repositories as well as
between toolkits (though the latter is still confounded by differences
in lower level cheminformatic features such as aromaticity models).

We have already discussed the development of an Open SMILES standard.
While much progress has been made towards a complete specification,
more remains to be done before this can be considered finished. After
that point, the next logical step would be to start work on a standard
for the SMARTS language, the extension to SMILES that specifies
patterns that match chemical substructures.

\section*{Open Data}

A considerable stumbling block in advocating the release of scientific
data as Open Data has been how exactly to define ``Open.'' A major step
forward was the launch in 2010 of the Panton Principles for Open Data
in Science \cite{WebPanton}. This formalises the idea that Open Data maximises the
possibility of reuse and repurposing, the fundamental basis
of how science works. These principles recommend that published data
be licensed explicitly, and preferably under CC0 (Creative Commons `No
Rights Reserved', also known as CCZero) \cite{WebCC0}. This license allows others to use the
data for any purpose whatsoever without any barriers. Other licenses
compatible with the Panton Principles include the
Open Data Commons Public Domain Dedication and Licence (PDDL), the
Open Data Commons Attribution License, and the
Open Data Commons Open Database License (ODbL).\cite{WebOpenData}

Despite this positive news, little chemical data has become
available from the traditional chemical fields of organic,
inorganic, solid state chemistry. Table~2 lists a few notable
exceptions: ChemPedia (a now discontinued crowd-sourcing project),
CrystalEye~\cite{WebCrystalEye},
and the Open Notebook Science Solubility
data~\cite{ONS2010}. There is also data available using licenses
not compatible with the Panton Principles, but where the user
is allowed to modify and redistribute the data. A new data
set in this category is the data from the ChEMBL database,
which is available under the Creative Commons Share-Alike
Attribution license~\cite{Overington2009}.

Importantly, publishing data as CC0 is becoming easier now that
websites are becoming available to simplify publishing data. Two
projects that can be mentioned in this context are
FigShare\cite{WebFigShare}, where the data behind unpublished figures
can be hosted, and Dryad\cite{WebDryad} where data behind publications
can be hosted. Initiatives like this make it possible to host small
amounts of data, and those combined are expected to become soon a
substantial knowledge base. 

\subsection*{The Blue Obelisk Data Repository (BODR)}

The Blue Obelisk has created a repository of key chemical
data in a machine-readable format~\cite{BODR}.
The BODR focuses on data that is commonly required for
chemistry software, and where there is a need to ensure that values
are standard between codes. Examples are atomic
masses and conversions between physical constants. These data
can be used by others for any purpose (for example, for entry into
Wikipedia or use in in-house software), and should lead
to an enhancement in the quality of community reference data.

\subsection*{NMRShiftDB}

NMRShiftDB \cite{nmrshiftdb, Steinbeck:2004do, Steinbeck2004} represents one of the earliest resources for Open
community-contributed data (first released in 2003). Research groups that 
measure NMR spectra or extract it from the literature can contribute that
information to
NMRShiftDB which provides an Open resource where
entries can be searched by chemical structure or properties
(especially peaks). Although it is difficult to
encourage large amounts of altruistic contribution (as happens with
Wikipedia), an alternative possible source of data could come from
linking
data capture with data publication. For example, the Blue Obelisk has
enough software that it is possible to create
a seamless chain for converting NMR structures in-house into
NMRShiftDB entries. If and when the chemistry community
encourages or requires semantic publication of spectra rather than
PDFs, it would be possible to populate NMRShiftDB rapidly
along the the lines of CrystalEye (see below).

\subsection*{CrystalEye}

CrystalEye~\cite{WebCrystalEye} is an example of cost-effective extraction of data from the
literature where this is published both
Openly and semantically. Software extracts Openly-published crystal
structures from a variety of scholarly journals, processes
them and then makes them available through a web interface.
It currently contains about 250,000 structures. CrystalEye serves as a model
for a high-value, high-quality Open data
resource, including the licensing of each component as
Panton-compatible Open data.

\section*{Other areas of activity}

While each Blue Obelisk project has its own website and point of
contact (typically a mailing list), because of the breadth of BO
projects it can be difficult for a newcomer to understand which of
them, if any, can best address a particular problem. To address this
issue, members of the Blue Obelisk established a Question \& Answer
website\cite{WebBOShapado} (see Figure~5).  This is a website in the
style of Stack Overflow\cite{WebStackOverflow} that encourages high quality answers (and
questions) through the use of a voting system. In the year since it
was established, over 200 users have registered, many of whom had no
previous involvement with the Blue Obelisk, showing that the Q\&A
website complements earlier existing channels of communication.

The rise of self-publishing and print-on-demand services has meant
that publishing a book is now as straightforward as uploading to an
appropriate website. Unlike the traditional publishing route where
books with projected low sales volume would be expensive,
websites such as Lulu\cite{WebLulu} allow the sale of low-priced books on
chemistry software, and books are now available for purchase
on Jmol~\cite{JmolBook}, the Chemistry Development Kit~\cite{CDKBook}
and Open Babel~\cite{Open BabelBook}.

%%%%%%%%%%%%%%%%%%%%%%
\section*{Conclusions}

We have shown that the Blue Obelisk has been very successful
in bringing together researchers and developers with common interests
in ODOSOS, leading to development of many useful resources freely
available to the chemistry community. But how best to engage with the
wider chemistry community outside of the Blue Obelisk remains an open
question. If the Blue Obelisk is truly to make an impact,
then an attempt must be made to reach beyond the subscribers to the
BO mailing list and blogs of members.

We hope to see this involvement between the Blue Obelisk and the wider
community grow in the future. To this end, we encourage the reader to
visit the Blue Obelisk website\cite{WebBlueObelisk}, send a message to our mailing list,
investigate related projects or read our blogs.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Authors contributions}
   Charles Darwin did all the work. The others stole the glory.


%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Acknowledgements}
  \ifthenelse{\boolean{publ}}{\small}{}
  Thanks to everyone.



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%
%%  Bmc_article.bst  will be used to                       %%
%%  create a .BBL file for submission, which includes      %%
%%  XML structured for BMC.                                %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %%
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %%
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


{\ifthenelse{\boolean{publ}}{\footnotesize}{\small}
 \bibliographystyle{bmc_article}  % Style BST file
  \bibliography{websites,paper} }     % Bibliography file (usually '*.bib' )

%%%%%%%%%%%

\ifthenelse{\boolean{publ}}{\end{multicols}}{}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

\section*{Figures}

 \subsection*{Figure 1 - Screenshot of Bioclipse using Jmol to
visualise a molecular surface}

 \subsection*{Figure 2 - Screenshot of Avogadro showing a depiction of
a carbon nanotube}

 \subsection*{Figure 3 - Screenshot of the MolGrabber 3D demo from
ChemDoodle Web Components}

  \subsection*{Figure 4 - Dependency diagram of some Blue Obelisk projects.}
Each block represents a project. Square blocks show Open Data, ovals are Open Source,
and diamonds are Open Standards.

  \subsection*{Figure 5 - Screenshot of the Blue Obelisk eXchange Question
    and Answer website.}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%
\section*{Tables}
  \subsection*{Table 1 - Blue Obelisk Open Source software projects}
    (Description if necessary XXXXXXXXXXXXXXX. Add citations to project names.)
      %% GeoffH - I think we should do this as multi-line, with the
      %% website below the Name
    \par \mbox{}
    \par
    \mbox{
     \begin{tabular}{|c|c|c|}
       \hline \textbf{Name} & \textbf{Website} & \textbf{Description} \\ \hline
        \multicolumn{3}{|c|}{\textbf{CML Tools}} \\ \hline
		\textbf{JUMBO \& CMLXOM} & http://sourceforge.net/projects/cml/ https://bitbucket.org/wwmm/cmlxom/ & Peter Murray-Rust et al. \\ \hline
		\multicolumn{3}{|c|}{\textbf{Cheminformatics Toolkits}} \\ \hline
        \textbf{Chemistry Development Kit (CDK)} \cite{Steinbeck2003, Steinbeck2006} & http://cdk.sf.net & XXXX  \\ \hline
        \textbf{Cinfony} & http://cinfony.googlecode.com & Noel O'Boyle, Python interface to toolkits \\ \hline
        \textbf{Indigo} & http://ggasoftware.com/opensource/indigo & GGA Software \\ \hline
        \textbf{Open Babel} & http://openbabel.org & Geoffrey Hutchison et al \\ \hline
        \textbf{RDKit} & http://rdkit.org & Greg Landrum \\ \hline
        \multicolumn{3}{|c|}{\textbf{Second Generation}} \\ \hline
        \textbf{Avogadro} & http://avogadro.openmolecules.net & XXX \\ \hline
        \multicolumn{3}{|c|}{\textbf{Web Applications}} \\ \hline
        \textbf{ChemDoodle Web Components} & http://web.chemdoodle.com & iChemLabs \\ \hline
        \textbf{Jmol} & http://jmol.org & Hanson \\ \hline
       \multicolumn{3}{|c|}{\textbf{Integration}} \\ \hline
        \textbf{CDK-Taverna} \cite{Kuhn:2010p4001} & http://cdktaverna.wordpress.com & Christoph Steinbeck, Workflow \\ \hline
		\textbf{Lensfield2} & https://bitbucket.org/sea36/lensfield2/ & Sam Adams et al. \\ \hline
        \textbf{A3} & ..  & .  \\ \hline
        \multicolumn{3}{|c|}{\textbf{Interconversion}} \\ \hline
        \textbf{OSRA} & http://osra.sf.net & Igor Filippov, Image to structure \\ \hline
        \textbf{OPSIN} & http://opsin.ch.cam.ac.uk & Daniel Lowe, Name to structure \\ \hline
		\textbf{JUMBO-Converters} & https://bitbucket.org/wwmm/jumbo-converters/ & Peter Murray-Rust et al. \\ \hline
		\textbf{CIFXOM} & https://bitbucket.org/wwmm/cifxom/ & Peter Murray-Rust et al. \cite{DayEtAl2011} \\ \hline
        \textbf{A3} & ..  & .  \\ \hline
        \multicolumn{3}{|c|}{\textbf{Structure Databases}} \\ \hline
        \textbf{Bingo}  & http://XXX.XXX.XXX & Dmitry Pavlov, Oracle-based Chemical Database Engine \\ \hline
        \textbf{MyChem}  & http://XXX.sf.net & Jerome Pansanel, MySQL-based Chemical Database Engine \\ \hline
        \textbf{OrChem} \cite{RijnbeekS10} & http://orchem.sf.net & Christoph Steinbeck, Oracle-based Chemical Database Engine \\ \hline
        \textbf{PGChem}  & http://XXX.sf.net & Ernst-Georg Schmid, PostgreSQL-based Chemical Database Engine \\ \hline
 	    \textbf{Chempound (Chem\#)} & https://bitbucket.org/chempound & Sam Adams et al. \\ \hline
	   \multicolumn{3}{|c|}{\textbf{Text mining}} \\ \hline
	    \textbf{OSCAR4} & https://bitbucket.org/wwmm/oscar4/ & Peter Murray-Rust et al. \\ \hline
		\textbf{ChemicalTagger} & http://chemicaltagger.ch.cam.ac.uk/ & Lezan Hawizy et al. \cite{HawizyEtAl2011} \\ \hline
       \multicolumn{3}{|c|}{\textbf{Other}} \\ \hline
        \textbf{cclib}  & http://cclib.sf.net & Python library for parsing and interpreting computational chemistry results \\
        \hline
%% GaussSum, QMForge
       \end{tabular}
      }
  \subsection*{Table 2 - Open Data in chemistry.}
    Overview of major open chemical data available under a license or waiver
    compatible with the Panton Principles.
    \par \mbox{}
    \par
    \mbox{
\begin{tabular}{|l|c|l|}
  %% \hline \multicolumn{3}{|c|}{My Table}\\ \hline
  \hline \textbf{Name} & \textbf{License/Waiver} & \textbf{Description} \\ \hline
  \textbf{ChemPedia} & CC0 & Crowd-sourced chemical names. Project discontinued. \\ \hline
  \textbf{CrystalEye}     & PPDL & Crystal structures from primary literature. \\ \hline
  \textbf{ONS Solubility} & CC0 & Solubility data for various solvents. \\ \hline
  \textbf{NMRShiftDB} & Public Domain & Database of NMR spectra of organic compounds. \\ \hline
\end{tabular}
      }



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Additional Files}
  \subsection*{Additional file 1 --- Sample additional file title}
    Additional file descriptions text (including details of how to
    view the file, if it is in a non-standard format or the file extension).  This might
    refer to a multi-page table or a figure.

  \subsection*{Additional file 2 --- Sample additional file title}
    Additional file descriptions text.


\end{bmcformat}
\end{document}







